src/Controllers/Training/leaveDemo.php

```diff
@@ -3,5 +3,5 @@
 use \MyRadio\MyRadio\URLUtils;
 use \MyRadio\ServiceAPI\MyRadio_Demo;
 
-MyRadio_Demo::getInstance($_REQUEST['demoid'])->leave();
-URLUtils::redirect($module, 'listDemos', ['msg' => 3]); // 3 means left... see listDemos.php
+$result = MyRadio_Demo::getInstance($_REQUEST['demoid'])->leave();
+URLUtils::redirect($module, 'listDemos', ['msg' => $result === 0 ? -1 : $result * -1]); // -1 means left... see listDemos.php
```

Comment:
The -n messages are somewhat confusing. I see why you're doing it (to make leaving distinguished from joining) but if you do that maybe http status code style grouping is better (101, 201, 301 etc.)? Even then is it necessary? Especially when your only cases are -1 and -3.

Reply:
Yeah I wasn't exactly happy about this when I made it - my reasoning is that both 3 codes are for the time limit (+3 for joining and -3 for leaving). I think I'll stick to it like this, because so far there's not enough to justify finer grouping in my head.